editor/src/clj/editor/app_view.clj

```diff
@@ -2619,3 +2623,52 @@ If you do not specifically require different script states, consider changing th
   (enabled? [] (disk-availability/available?))
   (run [app-view changes-view prefs workspace project]
        (ensure-exists-and-open-for-editing! shared-editor-settings/project-shared-editor-settings-proj-path app-view changes-view prefs project)))
+
+(defn get-linux-desktop-entry [launcher-path install-dir]
+  (str "[Desktop Entry]\n"
+       "Name=Defold\n"
+       "Comment=An out of the box, turn-key solution for multi-platform game development\n"
+       "Terminal=false\n"
+       "Type=Application\n"
+       "StartupWMClass=com.defold.editor.Start\n"
+       "Categories=Games;Development;Editor;\n"
+       "StartupNotify=true\n"
+       "Exec=" launcher-path "\n"
+       "Icon=" install-dir "/logo_blue.png\n"))
+
+(def ^:private xdg-desktop-menu-path
+  (delay
+    (when (util/is-linux?)
+      (try
+        (process/exec! "which" "xdg-desktop-menu")
+        (catch Throwable _)))))
+
+(handler/defhandler :create-desktop-entry :global
+  (active? [] (some? @xdg-desktop-menu-path))
+  (enabled? [] (and (system/defold-resourcespath) (system/defold-launcherpath)))
+  (run []
+       (let [xdg-desktop-menu @xdg-desktop-menu-path
+             install-dir (.getCanonicalFile (io/file (system/defold-resourcespath)))
+             launcher-path (.getCanonicalFile (io/file (system/defold-launcherpath)))
+             desktop-entry (get-linux-desktop-entry launcher-path install-dir)
+             desktop-entry-file (io/file install-dir "defold-editor.desktop")]
+         (try
```

Comment:
Great, also haven't thought about exceptions and the user feedback for this.

Reply:
`process/exec!` will throw if the exit code is not zero, we should probably document it.